commentedOutImport_checker.go

```diff
@@ -0,0 +1,75 @@
+package checkers
+
+import (
+	"go/ast"
+	"go/token"
+	"regexp"
+
+	"github.com/go-lintpack/lintpack"
+	"github.com/go-lintpack/lintpack/astwalk"
+)
+
+func init() {
+	var info lintpack.CheckerInfo
+	info.Name = "commentedOutImport"
+	info.Tags = []string{"style", "experimental"}
+	info.Summary = "Detects commented-out imports"
+	info.Before = `
+import (
+	"fmt"
+	//"os"
+)`
+	info.After = `
+import (
+	"fmt"
+)`
+
+	collection.AddChecker(&info, func(ctx *lintpack.CheckerContext) lintpack.FileWalker {
+		return &commentedOutImportChecker{
+			ctx:            ctx,
+			importStringRE: regexp.MustCompile(`(?m)^(?://|/\*)?\s*"([a-zA-Z0-9_/]+)"\s*(?:\*/)?$`),
+		}
+	})
+}
+
+type commentedOutImportChecker struct {
+	astwalk.WalkHandler
+	ctx *lintpack.CheckerContext
+
+	importStringRE *regexp.Regexp
+}
+
+func (c *commentedOutImportChecker) WalkFile(f *ast.File) {
+	// TODO(Quasilyte): handle commented-out import spec,
+	// for example: // import "errors".
```

Comment:
Leftover?

Reply:
No, matching of commented-out import decls is not implemented (yet?).
Not hard to do though.
Can be a "good first issue" for someone.

Reply:
Can you create the issue for that?

Reply:
Yes, but probably after we merge into `go-critic`.